### DISCUSSION TITLE
Will a new language ever be introduced?

### DISCUSSION THREAD

**Discussion by RuboGubo** [cc-tweaked/CC-Tweaked#673] [Q&A] [answered]
## I was wondering if a new language(such as python) would ever be introduced. 
I ask this because the documentation is not great for lua while for other languages it is far better potential making the mod more accessible.

**Comment by Lupus590** [comment, marked as answer]
This has been asked a few times and although it would be relatively easy to integrate a new language engine (I believe it just needs to implement the right interface), one would have to rewrite the entire ROM in the new language as well as make sure that third-party peripherals are compatible with all languages bundled with the mod. On top of this, a new language would fracture the already small community and make support more difficult.

One could fork the mod and change the language, making a CC: Python or whatever, if you change the mod ID then you could have CC:P run along with CC:T (I think you'd need to change the mod ID, I'm still new to forge mods and java in general).

On a tangential note, you could try transpiling: https://github.com/hengestone/lua-languages

See also: https://github.com/SquidDev-CC/CC-Tweaked/issues/547#issuecomment-701237857

**Comment by SquidDev** [comment]
> I ask this because the documentation is not great for lua

Just curious as to where you've found the documentation lacking? When I first picked it up I didn't find it too bad, but I'm probably more generous in overlooking flaws.
